src/main/java/org/sagebionetworks/web/client/widget/entity/file/FileDownloadMenuItemViewImpl.java

```diff
@@ -52,46 +42,59 @@ public FileDownloadMenuItemViewImpl(PortalGinInjector ginInjector) {
   public void clear() {}
 
   private void clearClickHandlers() {
-    if (downloadLinkHandlerRegistration != null) {
-      downloadLinkHandlerRegistration.removeHandler();
-    }
-    downloadLink.setHref("#");
+    this.actionMenu.setActionHref(Action.DOWNLOAD_FILE, "#");
   }
 
   @Override
   public void setIsSFTPDownload() {
-    String href = downloadLink.getHref();
+    String oldHref = this.href;
     clearClickHandlers();
-    downloadLink.setHref(href);
-    downloadLinkHandlerRegistration =
-      downloadLink.addClickHandler(sftpDownloadClickHandler);
+    this.href = oldHref;
+    actionMenu.setActionHref(Action.DOWNLOAD_FILE, this.href);
+    actionMenu.addActionListener(
+      Action.DOWNLOAD_FILE,
+      (Action action, ReactMouseEvent event) ->
+        sftpDownloadClickHandler.onClick(event)
+    );
```

Comment:
These handlers are basically the same, so we can have some confidence that they still work. They are now just registered with the action menu.